src/main/java/org/example/learningsystem/btp/destinationservice/service/CloudDestinationServiceImpl.java

```diff
@@ -29,17 +33,29 @@ public class CloudDestinationServiceImpl implements DestinationService {
     @Override
     @Retryable(retryFor = Unauthorized.class, maxAttempts = 2)
     public DestinationDto getByName(String name) {
-        return tryToGetDestination(name);
+        var tenantTokenUrl = getTenantTokenUrl();
+        if (tenantTokenUrl.isPresent()) {
+            try {
+                log.info("Trying to retrieve destination {} for tenant url {}", name, tenantTokenUrl.get());
+                return tryToGetDestination(tenantTokenUrl.get(), name);
+            } catch (NotFound e) {
+                log.info("Destination {} wasn't found for current tenant", name);
+            }
+        }
```

Comment:
it's better to split this into 2 methods, where you will try to find destination in tenant subaccount and only then on provider